mlflow/models/dependencies_schemas.py

```diff
@@ -56,11 +52,27 @@ def set_retriever_schema(
                 other_columns=["title"],
             )
     """
-    globals()[_RETRIEVER_PRIMARY_KEY] = primary_key
-    globals()[_RETRIEVER_TEXT_COLUMN] = text_column
-    globals()[_RETRIEVER_DOC_URI] = doc_uri
-    globals()[_RETRIEVER_OTHER_COLUMNS] = other_columns or []
-    globals()[_RETRIEVER_NAME] = name
+    retriever_schema = globals().get(DependenciesSchemasType.RETRIEVERS.value, [])
+
+    # Check if a retriever schema with the same name already exists
+    if any(schema["name"] == name for schema in retriever_schema):
+        # reset if there is an error to clear the global state for next run
+        _clear_retriever_schema()
+        raise MlflowException(
```

Comment:
Since there are no methods to delete the retriever_schema, once you set it you cannot update/overwrite it.
Could we please remove this exception and overwrite any existing schema with the same name, like it worked before?